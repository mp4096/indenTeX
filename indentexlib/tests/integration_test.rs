use insta::assert_display_snapshot_matches;

#[cfg(test)]
fn transpile_from_file(test_filename: &str) -> String {
    let manifest_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
    let test_case_path: std::path::PathBuf = [&manifest_dir, "tests", "assets", test_filename]
        .iter()
        .collect();
    let test_case_file = std::fs::File::open(test_case_path).unwrap();
    let test_case_buf = std::io::BufReader::new(test_case_file);
    let lines = indentexlib::preprocessing::read_and_trim_lines(test_case_buf).unwrap();

    let default_options = indentexlib::TranspileOptions {
        flatten_output: false,
        prepend_do_not_edit_notice: true,
    };

    indentexlib::transpile(lines, &default_options)
}

#[test]
fn transpile_corner_cases() {
    assert_display_snapshot_matches!(
        "corner_cases",
        transpile_from_file("corner_cases.inden.tex")
    );
}

#[test]
fn transpile_envs() {
    assert_display_snapshot_matches!("envs", transpile_from_file("envs.inden.tex"));
}

#[test]
fn transpile_large_corpus() {
    assert_display_snapshot_matches!(
        "large_corpus",
        transpile_from_file("large_corpus.inden.tex")
    );
}

#[test]
fn transpile_list_like() {
    assert_display_snapshot_matches!("list_like", transpile_from_file("list_like.inden.tex"));
}

#[test]
fn transpile_mixed_tabs() {
    assert_display_snapshot_matches!("mixed_tabs", transpile_from_file("mixed_tabs.inden.tex"));
}

#[test]
fn transpile_single_line_cmds() {
    assert_display_snapshot_matches!(
        "single_line_cmds",
        transpile_from_file("single_line_cmds.inden.tex")
    );
}
