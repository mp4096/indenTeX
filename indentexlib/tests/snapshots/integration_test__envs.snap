---
created: "2019-06-23T23:00:21.101271654Z"
creator: insta@0.8.1
source: indentexlib/tests/integration_test.rs
expression: "transpile_from_file(\"envs\")"
---
% ============================================================== %
%                                                                %
% THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. %
%                                                                %
% ============================================================== %
\begin{equation*}
  \label{eq:test}
  a + b
\end{equation*}

\begin{tikzpicture}[x = 2 cm]
  \draw (0, 0) -- (1, 1);
\end{tikzpicture}

Comments should be handled correctly:
\begin{equation} % test
  a + b
\end{equation}
\begin{remark}[test percent escaping \%] % baz
  foo bar
\end{remark}

This should be converted correctly (also check if all environments are closed at the EOF):
\begin{a}
  \begin{b}
    \begin{c}
      \begin{d}
        \begin{e}
          \begin{f}
            \begin{g}
              \begin{h}
                \begin{i}
                  foobar
                \end{i}
              \end{h}
            \end{g}
          \end{f}
        \end{e}
      \end{d}
    \end{c}
  \end{b}
\end{a}

